09-features.md

```diff
@@ -45,6 +45,7 @@ The Context column decodes as follows:
 | 46/47 | `option_scid_alias`              | Supply channel aliases for routing                        | IN       |                   | [BOLT #2][bolt02-channel-ready]   |
 | 48/49 | `option_payment_metadata` | Payment metadata in tlv record | 9 | | [BOLT #11](11-payment-encoding.md#tagged-fields)
 | 50/51 | `option_zeroconf`                | Understands zeroconf channel types                        | IN       | `option_scid_alias` | [BOLT #2][bolt02-channel-ready]   |
+| 56/57 | `trampoline_routing`              | This node supports trampoline routing                     | IN9      | `basic_mpp`               | [BOLT #4](bolt04-trampoline)                                          |
```

Comment:
Will need to add this feature bit for BOLT 12 invoices, and adapt the spec for the lack of payment secret in them :)

Reply:
Indeed! That should be quite easy though, since there are two "layers" of payment secrets:

- the real payment secret that comes from the invoice and is used by the sender in the trampoline onion: this will simply be replaced by usage of a blinded path
- the payment secrets generated by each trampoline node for the payment to the next trampoline node (in the _outer_ onion): those will stay, since intermediate trampoline nodes don't need to know that blinded paths are being used, except maybe for the last trampoline node before the recipient

I should also describe how blinded paths can be used on the trampoline route, which is very interesting when paying public nodes. I'll wait for the blinded path PR to be merged though before doing all that!

Reply:
I have added details about the trampoline-to-blinded paths case in https://github.com/lightning/bolts/pull/836/commits/7589dd8e896eb2e3d206a05abb8a8c654057b693

Since Bolt 12 is scattered across a few PRs and pending rebases, it's a bit messy to reference the right TLV types: it will need some clean-up once Bolt 12 is merged. But the test vector should let you verify the exact encoding!